### DISCUSSION TITLE
Would it be possible to get some API for dropping files from a file system into a custom view?

### DISCUSSION THREAD

**Discussion by Luke-Callaghan23** [microsoft/vscode-discussions#438] [Extension Development QnA] [answered]
Hi, I'm wondering if it's possible to get some sort of API for handling dropped files from some external source (a file system window) into a custom view, kind of like how in the Explorer view of default vscode, you can drop files from your OS's file system and they will get copied into the workspace of vscode.  The extension I'm developing has a custom file explorer and it would be helpful for me to be able to drop files from the file system into that view and handle copying those files from their source location onto my custom file system.

As far as I can tell, the only thing that allows *similar* functionality right now is `vscode.languages.registerDocumentDropEditProvider`, but this only provides for a rather niche use-case of dropping files into the **editor** window when shift is held down.  I'd like some API for handling when a file is dropped into the tree view I am creating, without the need for shift to be held down.

Any help would be great.

Thanks

**Comment by isidorn** [comment, marked as answer]
This is a fair request but I think it is not on the plan. I suggest to file a new API feature request in the VS Code repo https://github.com/microsoft/vscode/issues
